spec/message.abnf

```diff
@@ -27,10 +27,10 @@ function = (":" / "+" / "-") name
 option = name [s] "=" [s] (literal / variable)
 
 ; reserved keywords are always lowercase
-input = %x69.6E.70.75.74  ; "input"
-local = %x6C.6F.63.61.6C  ; "local"
-match = %x6D.61.74.63.68  ; "match"
-when  = %x77.68.65.6E     ; "when"
+input = %s"input"
+local = %s"local"
+match = %s"match"
+when  = %s"when"
```

Comment:
Why are these variables now needed? Couldn't their string values be used directly in the rules now referring to these variables?

Reply:
The `%s` is what makes the value case-sensitive. Plain strings are, for back-compat, case insensitive by default.

Reply:
I think @eemeli was asking why not use `%s"input"` inline instead of indirectly through a named production:

```diff
-input-declaration = input [s] variable-expression
+input-declaration = %s"input" [s] variable-expression
```

One reason to not do it this way is that it would make it harder to apply the workaround mentioned in syntax.md for tools that dont' support 7405.

Reply:
Yes, that's why I did it that way. 

We might go back and make the change later, after we decide what to do with the code-mode sigil.